native/src/main/java/io/ballerina/lib/avro/serialize/visitor/SerializeVisitor.java

```diff
@@ -97,39 +98,32 @@ private Object serializeField(Schema schema, Object fieldData) throws Exception
 
     @Override
     public Object visit(PrimitiveSerializer primitiveSerializer, Object data) throws Exception {
-        switch (primitiveSerializer.getSchema().getType()) {
-            case INT -> {
-                return ((Long) data).intValue();
-            }
-            case FLOAT -> {
-                return ((Double) data).floatValue();
-            }
+        return switch (primitiveSerializer.getSchema().getType()) {
+            case INT -> ((Long) data).intValue();
+            case FLOAT -> ((Double) data).floatValue();
```

Comment:
I'm not sure if these casts are always safe, as we try to cast based on the schema type in here. Shouldn't we add `instanceof` checks and do proper error handling in that case?